File: introductiontounix2/sol/solution.java
import java.util.Scanner;
import java.util.Map;
import java.util.HashMap;

public class solution {
    public static void main(String[] args) {
        // Create the scanner
        Scanner scnr = new Scanner(System.in);

        // Read the number of commands
        int num_test = Integer.parseInt(scnr.nextLine());
        // Create a HashMap to store unique file names and contents
        Map<String, String> files = new HashMap<>();
        // Read commands until cat
        while (true) {
            // Read the command
            String command = scnr.nextLine();
            if (command.substring(0, 2).equals("vi")) { // Command is vi
                // Get the file name
                String fileName = command.substring(3);
                // Reset the file
                files.put(fileName, "");

                // Append lines to the file contents until :wq
                String line = scnr.nextLine();
                while (!line.equals(":wq")) {
                    // Update file's contents
                    files.put(fileName, files.get(fileName) + line + "\n");
                    // Read the next line
                    line = scnr.nextLine();
                }
            } else { // Command is cat
                // Print the contents of the file
                System.out.println(files.get(command.substring(4)));

                // Close the scanner
                scnr.close();
                // Break out of the loop
                break;
            }
        }
    }
}

File: introductiontounix2/sol/solution.py
def solve(commands):
    contents_dict = {}
    i = 0
    file_name = ""
    contents = ""
    reading_contents = False
    while i < len(commands):
        command = commands[i]
        if i < len(commands) - 1:
            if reading_contents:
                if command == ":wq":
                    reading_contents = False
                    contents_dict[file_name] = contents
                    contents = ""
                else:
                    contents += command + "\n"
            else:
                file_name = command[3:]
                reading_contents = True
        else:
            file_name = command[4:]
            return contents_dict[file_name]
        i += 1

commands = []

for i in range(int(input())):
    commands.append(input())

print(solve(commands))